### ISSUE TITLE
Kak-lsp not working with Julia

### ISSUE THREAD

**Issue by egolep** [kak-lsp/kak-lsp#509]
Since monday, I have no way to use kak-lsp with Julia.
I keep getting the error I paste at the end of this issue. I tried to delete and recreate the .julia folder, not installing any package, installing some packages, only one package etc.
I think it's a problem with kak-lsp since if I try to instantiate the language server inside a Julia REPL I get no error at all, and the "0-element" error looks like a default behavior that takes for granted the fact that arrays start with index 0, but in Julia (like ForTran) arrays are indexed starting from 1.
I tried kak-lsp with Rust and it works flawlessly

Aug 16 02:56:22.804 INFO Starting main event loop, module: kak_lsp::session:28
Aug 16 02:56:29.772 INFO Starting Language server `julia --startup-file=no --history-file=no -e         using LanguageServer;
        using Pkg;
        import StaticLint;
        import SymbolServer;
        import REPL;
        env_path = dirname(Pkg.Types.Context().env.project_file);
        server = LanguageServer.LanguageServerInstance(stdin, stdout, env_path, "");
        server.runlinter = true;
        run(server);
    `, module: kak_lsp::language_server_transport:20
Aug 16 02:56:54.437 ERRO Language server error: ERROR: BoundsError: attempt to access 0-element Vector{Any} at index [1:12]
Stacktrace:
 [1] checkbounds
   @ ./abstractarray.jl:616 [inlined]
 [2] getindex
   @ ./array.jl:807 [inlined]
 [3] request_julia_config(server::LanguageServerInstance, conn::JSONRPC.JSONRPCEndpoint{Base.PipeEndpoint, Base.PipeEndpoint})
   @ LanguageServer ~/.julia/packages/LanguageServer/JrIEf/src/requests/workspace.jl:113
 [4] initialized_notification(params::LanguageServer.InitializedParams, server::LanguageServerInstance, conn::JSONRPC.JSONRPCEndpoint{Base.PipeEndpoint, Base.PipeEndpoint})
   @ LanguageServer ~/.julia/packages/LanguageServer/JrIEf/src/requests/init.jl:195
 [5] (::LanguageServer.var"#96#97"{typeof(LanguageServer.initialized_notification), LanguageServerInstance})(conn::JSONRPC.JSONRPCEndpoint{Base.PipeEndpoint, Base.PipeEndpoint}, params::LanguageServer.InitializedParams)
   @ LanguageServer ~/.julia/packages/LanguageServer/JrIEf/src/languageserverinstance.jl:252
 [6] dispatch_msg(x::JSONRPC.JSONRPCEndpoint{Base.PipeEndpoint, Base.PipeEndpoint}, dispatcher::JSONRPC.MsgDispatcher, msg::Dict{String, Any})
   @ JSONRPC ~/.julia/packages/JSONRPC/yu0G3/src/typed.jl:67
 [7] run(server::LanguageServerInstance)
   @ LanguageServer ~/.julia/packages/LanguageServer/JrIEf/src/languageserverinstance.jl:361
 [8] top-level scope
   @ none:9
, module: kak_lsp::language_server_transport:58
Aug 16 02:56:54.437 INFO Waiting for Messages to language server to finish..., module: kak_lsp::thread_worker:19
Aug 16 02:56:55.437 INFO ... Messages to language server terminated with ok, module: kak_lsp::thread_worker:21
Aug 16 02:56:55.437 INFO Waiting for Messages from language server to finish..., module: kak_lsp::thread_worker:19
Aug 16 02:56:55.437 INFO ... Messages from language server terminated with ok, module: kak_lsp::thread_worker:21
Aug 16 02:56:55.437 INFO Waiting for Language server errors to finish..., module: kak_lsp::thread_worker:19
Aug 16 02:56:55.437 INFO ... Language server errors terminated with ok, module: kak_lsp::thread_worker:21

**Comment by krobelus** [comment]
Which version of kak-lsp are you using?
I hit the [same error](https://github.com/kak-lsp/kak-lsp/pull/502#issuecomment-894782284) on latest master.
OTOH it works with kak-lsp 10.0.0 with this Dockerfile

```Dockerfile
FROM docker.io/archlinux:latest

RUN pacman -Sy --noconfirm julia kakoune kak-lsp

RUN groupadd -g 1000 user \
  && useradd -p $(openssl passwd -1 Password) -d /home/user -m -u 1000 -g 1000 user

USER user

RUN mkdir -p ~/.config/kak/ && printf '%s\n' >~/.config/kak/kakrc \
	'eval %sh{kak-lsp --kakoune -s $kak_session}' \
	'set global lsp_cmd "kak-lsp -s %val{session} -vvvv --log /home/user/log"' \
	'lsp-enable'

WORKDIR /home/user
RUN julia -e 'import Pkg; Pkg.add("LanguageServer"); Pkg.add("StaticLint"); Pkg.add("SymbolServer")'
USER user

RUN mkdir ~/.config/kak-lsp
COPY kak-lsp.toml /home/user/.config/kak-lsp/kak-lsp.toml

RUN printf 'a = 1\nb = a\nprint(123)\n' > test.jl
```

**Comment by krobelus** [comment]
Ok this was caused by cc9e2925774ea0c14189264004444c8d12a5ac66. I guess that's still a good commit because it makes us find bugs like this.

**Comment by egolep** [comment]
Yeah I'm using the latest version available using plug.kak

Could I install kak-lsp with a different method and solve the problem or should I just wait for a new commit?

**Comment by krobelus** [comment]
It's fixed by 2ad57f4, so plug.kak should show that there is an update available.

**Comment by egolep** [comment]
It now works flawlessly. Many thanks!
